server/test/durablejobs/DurableJobRunnerTest.java

```diff
@@ -0,0 +1,167 @@
+package durablejobs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import annotations.BindingAnnotations;
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import models.PersistedDurableJob;
+import org.junit.Before;
+import org.junit.Test;
+import play.api.inject.BindingKey;
+import repository.PersistedDurableJobRepository;
+import repository.ResetPostgres;
+
+public class DurableJobRunnerTest extends ResetPostgres {
+
+  private DurableJobRunner durableJobRunner;
+  private DurableJobRegistry durableJobRegistry;
+
+  @Before
+  public void setUp() {
+    Config config =
+        ConfigFactory.parseMap(
+            ImmutableMap.of(
+                "durable_jobs.job_timeout_minutes", 0,
+                "durable_jobs.poll_interval_seconds", 0));
+
+    durableJobRegistry = new DurableJobRegistry();
+
+    durableJobRunner =
+        new DurableJobRunner(
+            config,
+            instanceOf(DurableJobExecutionContext.class),
+            durableJobRegistry,
+            instanceOf(PersistedDurableJobRepository.class),
+            () ->
+                instanceOf(
+                    new BindingKey<>(LocalDateTime.class)
+                        .qualifiedWith(BindingAnnotations.Now.class)),
+            instanceOf(ZoneId.class));
+  }
+
+  @Test
+  public void runJobs_timesOut() {
```

Comment:
**suggestion:** Add another test where the timeout isn't 0?

Reply:
The timeout is 0 specifically for test mode so the times can be much shorter